vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc

```diff
@@ -130,14 +130,28 @@ void Wallet::DisconnectWallet(
       const std::string& wallet_type,
       ledger::ResultCallback callback) {
   if (wallet_type == constant::kWalletUphold) {
-    ledger_->uphold()->DisconnectWallet(true);
-    callback(type::Result::LEDGER_OK);
+    promotion_server_->delete_claim()->Request(
+        constant::kWalletUphold, [this, callback](const type::Result result) {
+          if (result != type::Result::LEDGER_OK) {
```

Comment:
Question: what should we do on failure here? If we bail, then will the user be stuck in a connected state that they can't get out of? Is that what we want?